core/src/main/java/bisq/core/api/CoreWalletsService.java

```diff
@@ -185,6 +196,37 @@ void sendBsq(String address,
         }
     }
 
+    TxFeeRateInfo getTxFeeRate() {
+        try {
+            CompletableFuture<Void> feeRequestFuture = CompletableFuture.runAsync(feeService::requestFees);
+            feeRequestFuture.get();  // Block until async fee request is complete.
+        } catch (InterruptedException | ExecutionException e) {
+            throw new IllegalStateException("could not request fees from fee service.", e);
+        }
+
+        Coin stdTxFeeRate = feeService.getTxFeePerVbyte();
+        Coin customTxFeeRate = preferences.isUseCustomWithdrawalTxFee()
+                ? Coin.valueOf(preferences.getWithdrawalTxFeeInVbytes())
+                : Coin.NEGATIVE_SATOSHI;
+
+        return new TxFeeRateInfo(stdTxFeeRate.value, customTxFeeRate.value);
+    }
+
+    TxFeeRateInfo setTxFeeRatePreference(long txFeeRate) {
+        if (txFeeRate <= 0)
```

Comment:
We also don't want to create txs with fees < 1 sat/b, but perhaps that doesn't have to be checked here.

Reply:
Doing this check here makes the CLI's error msg stylistically consistent with others.  

This brings up a general issue that is going to need quite a bit of work in the API:  displaying validation error messages coming from the server's core, in the CLI.    But for now, getting all the v1 methods working for correct parameters is the short term priority.

Reply:
Agreed, not an issue for now, good to think of how to handle it though as you say.